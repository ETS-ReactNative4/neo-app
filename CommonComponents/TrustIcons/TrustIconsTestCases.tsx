import React from "react";

import { ITestCase } from "../../TypeInterfaces/TestCases/ITestCases";
import { View, StyleSheet } from "react-native";
import TrustIcons from "./TrustIcons";

const data: ITrustIconsData[] = [
  {
    logo: "https://i.imgur.com/TNnGL9a.png",
    text: "4.8/5 Based on 1200+ ratings"
  },
  {
    logo: "https://i.imgur.com/iRK9LRd.png",
    text: "4.5/5 Based on 300+ ratings"
  },
  {
    logo: "https://i.imgur.com/jWlTJvq.png",
    text: "Accredited Agent"
  }
];

export interface ITrustIconsData {
  logo: string;
  text: string;
}

interface TrustIconsWrapperProps {
  TrustIconsData: ITrustIconsData[];
}

const styles = StyleSheet.create({
  wrapper: {
    marginVertical: 24,
    flexDirection: "row",
    justifyContent: "space-between",
    paddingHorizontal: 24
  }
});

const TrustIconsWrapper = ({ TrustIconsData }: TrustIconsWrapperProps) => {
  return (
    <View style={styles.wrapper}>
      {TrustIconsData.map((item, index) => {
        return (
          <TrustIcons key={index} image={{ uri: item.logo }} text={item.text} />
        );
      })}
    </View>
  );
};

const TrustIconsTestCases: ITestCase[] = [
  {
    title: "Partner Section",
    Component: (
      <TrustIcons
        image={{ uri: "https://i.imgur.com/TNnGL9a.png" }}
        text={"4.8/5 Based on 1200+ ratings"}
      />
    )
  },
  {
    title: "Multiple Partner Section",
    Component: <TrustIconsWrapper TrustIconsData={data} />
  }
];

export default TrustIconsTestCases;
