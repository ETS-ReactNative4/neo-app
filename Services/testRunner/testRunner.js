import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
