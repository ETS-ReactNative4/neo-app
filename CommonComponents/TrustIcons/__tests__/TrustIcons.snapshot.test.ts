import TrustIconsTestCases from "../TrustIconsTestCases";
import testRunner from "../../../Services/testRunner/testRunner";

TrustIconsTestCases.map(testRunner);
