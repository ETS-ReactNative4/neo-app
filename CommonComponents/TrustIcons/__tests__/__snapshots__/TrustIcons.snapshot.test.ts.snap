// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Partner Section 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 24,
      "paddingHorizontal": 24,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "width": 80,
        },
        undefined,
      ]
    }
  >
    <Image
      onError={[Function]}
      resizeMode="contain"
      source={
        Object {
          "uri": "https://i.imgur.com/TNnGL9a.png",
        }
      }
      style={
        Object {
          "height": 55,
          "width": 55,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "rgba(119, 119, 119, 1)",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 12,
          "lineHeight": 16,
          "textAlign": "center",
        }
      }
    >
      4.8/5 Based on 1200+ ratings
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "width": 80,
        },
        undefined,
      ]
    }
  >
    <Image
      onError={[Function]}
      resizeMode="contain"
      source={
        Object {
          "uri": "https://i.imgur.com/iRK9LRd.png",
        }
      }
      style={
        Object {
          "height": 55,
          "width": 55,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "rgba(119, 119, 119, 1)",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 12,
          "lineHeight": 16,
          "textAlign": "center",
        }
      }
    >
      4.5/5 Based on 300+ ratings
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "width": 80,
        },
        undefined,
      ]
    }
  >
    <Image
      onError={[Function]}
      resizeMode="contain"
      source={
        Object {
          "uri": "https://i.imgur.com/jWlTJvq.png",
        }
      }
      style={
        Object {
          "height": 55,
          "width": 55,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "rgba(119, 119, 119, 1)",
          "fontFamily": "SourceSansPro-Regular",
          "fontSize": 12,
          "lineHeight": 16,
          "textAlign": "center",
        }
      }
    >
      Accredited Agent
    </Text>
  </View>
</View>
`;

exports[`Partner Section 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "width": 80,
      },
      undefined,
    ]
  }
>
  <Image
    onError={[Function]}
    resizeMode="contain"
    source={
      Object {
        "uri": "https://i.imgur.com/TNnGL9a.png",
      }
    }
    style={
      Object {
        "height": 55,
        "width": 55,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "rgba(119, 119, 119, 1)",
        "fontFamily": "SourceSansPro-Regular",
        "fontSize": 12,
        "lineHeight": 16,
        "textAlign": "center",
      }
    }
  >
    4.8/5 Based on 1200+ ratings
  </Text>
</View>
`;
